src/CashuWallet.ts

```diff
@@ -309,6 +344,36 @@ class CashuWallet {
 		};
 	}
 
+	/**
+	 * Regenerates
+	 * @param count? set number of blinded messages that should be generated
+	 * @param startIndex optionally set starting point for count (default is 0)
+	 * @returns proofs and newKeys if they have changed
+	 */
+	async restore(
+		count: number,
+		startIndex = 0,
+		keysetId?: string
+	): Promise<{ proofs: Array<Proof>; newKeys?: MintKeys }> {
+		if (!this._seed) {
+			throw new Error('CashuWallet must be initialized with mnemonic to use restore');
+		}
+		const amounts = Array(count).fill(0);
+		const { blindedMessages, rs, secrets } = this.createBlindedMessages(amounts, startIndex, keysetId);
```

Comment:
What is confusing me here are these `amounts`. Aren't these really indexes for deterministic secrets?